pkg/ui/progressbar.go

```diff
@@ -0,0 +1,49 @@
+package ui
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"strings"
+
+	"github.com/snyk/go-application-framework/pkg/utils"
+)
+
+const (
+	barCharacter    = "="
+	currentPosition = ">"
+	barWidth        = 50
+	clearLine       = "\r\033[K"
+)
+
+type ProgressBar interface {
+	SetProgress(progress float64) error
+	Clear() error
+}
+
+func newProgressBar(writer io.Writer) *consoleProgressBar {
+	return &consoleProgressBar{writer: writer}
+}
+
+type consoleProgressBar struct {
+	writer io.Writer
+}
+
+func (p consoleProgressBar) SetProgress(progress float64) error {
```

Comment:
Question: Could we add a Description in addition to the value? So that use cases like
> Analyzing file 50%
can be enabled.

Reply:
Done. I add the title at the end though (so it won't move the bar horizontally)